### PR TITLE
HCAP-660 Refactor ParticipantTable UseEffect, columns, tabs, context 

### PR DESCRIPTION
**Pull request by arranfw**
https://freshworks.atlassian.net/browse/HCAP-660

### CODE REVIEW

File: client/src/providers/AuthContext.js [hunk 1]
```diff
@@ -2,20 +2,25 @@ import React, { useReducer } from 'react';
...
+        ...state,
         user: null,
         isLoading: true,
       };
     case USER_LOADED:
       return {
+        ...state,
         user: action.payload,
         isLoading: false,
+        permissionRole: rolePriority.find((role) => action.payload?.roles.includes(role)),
```
Comment (arranfw) [inline client/src/providers/AuthContext.js:23]: find the first element in above array that matches one of the user's roles

File: client/src/providers/AuthContext.js [hunk 2]
```diff
@@ -2,20 +2,26 @@ import React, { useReducer } from 'react';
...
         user: null,
         isLoading: true,
       };
     case USER_LOADED:
       return {
+        ...state,
         user: action.payload,
         isLoading: false,
+        // find first role in above array that is included in the user object (highest permission role)
+        permissionRole: rolePriority.find((role) => action.payload?.roles.includes(role)),
```
Comment (fwpushan) [inline client/src/providers/AuthContext.js:24]: Nice

**Review by fwpushan — DISMISSED** [review]

File: client/src/providers/AuthContext.js [hunk 3]
```diff
@@ -2,20 +2,26 @@ import React, { useReducer } from 'react';
 
 const AuthContext = React.createContext();
 
+const rolePriority = ['superuser', 'ministry_of_health', 'health_authority', 'employer'];
```
Comment (dbayly-freshworks) [inline client/src/providers/AuthContext.js:5]: NIT,  but maybe this could go in a constants file

**Review by dbayly-freshworks — DISMISSED** [review]

File: client/src/providers/AuthContext.js (hunk 3 above)
Comment (arranfw) [reply client/src/providers/AuthContext.js:5]: Ahh good nit you're right

**Review by fwpushan — APPROVED** [review]

File: client/src/constants/participantTableConstants.js [hunk 4]
```diff
@@ -98,3 +67,217 @@ export const defaultTableState = {
...
+  preferredLocation: { id: 'preferredLocation', name: 'Preferred Region(s)', sortOrder: 9 },
+  distance: { id: 'distance', name: 'Site Distance', sortOrder: 10 },
+  interested: { id: 'interested', name: 'Interest', sortOrder: 11 },
+  nonHCAP: { id: 'nonHCAP', name: 'Non-HCAP', sortOrder: 12 },
+  crcClear: { id: 'crcClear', name: 'CRC Clear', sortOrder: 13 },
+  callbackStatus: { id: 'callbackStatus', name: 'Callback Status', sortOrder: 14 },
+  userUpdatedAt: { id: 'userUpdatedAt', name: 'Last Updated', sortOrder: 15 },
+  engage: { id: 'engage', name: null, sortOrder: 16 },
+  edit: { id: 'edit', name: null, sortOrder: 17 },
+  siteName: { id: 'siteName', name: 'Site Name', sortOrder: 18 },
```
Comment (dbayly-freshworks) [inline client/src/constants/participantTableConstants.js:109]: Looks like you'll need an 'Archive' column here. 

`archive: {id:'archive',name:null, sortOrder:19}`

File: client/src/constants/participantTableConstants.js [hunk 5]
```diff
@@ -98,3 +67,217 @@ export const defaultTableState = {
...
+      lastName,
+      firstName,
+      postalCodeFsa,
+      phoneNumber,
+      emailAddress,
+      preferredLocation,
+      distance,
+      siteName,
+      nonHCAP,
+      userUpdatedAt,
```
Comment (dbayly-freshworks) [inline client/src/constants/participantTableConstants.js:280]: And here.

File: client/src/constants/participantTableConstants.js [hunk 6]
```diff
@@ -98,3 +67,217 @@ export const defaultTableState = {
...
```
Comment (dbayly-freshworks) [inline client/src/constants/participantTableConstants.js:222]: Put `archive` here. It should fix the tests.

**Review by dbayly-freshworks — CHANGES_REQUESTED** [review]
Looks good. Added some comments on how to fix the tests.

**Review by fwpushan — APPROVED** [review]
👍

**Review by Zenen-freshworks — APPROVED** [review]
nothing is jumping out to me, looks good 👍

**Pull request merged**